### ISSUE TITLE
OverloadMethodsDeclarationOrderCheck: split top-level overloads not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20228]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/coding/overloadmethodsdeclarationorder.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="OverloadMethodsDeclarationOrder"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputOverloadMethodsCompact.java <<'EOF'
void process(int x) { }
void helper() { }
void process(String s) { }
void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml InputOverloadMethodsCompact.java

Starting audit...
Audit done.
```

Same code inside an ordinary class:

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > OrdinaryOverloadMethods.java <<'EOF'
public class OrdinaryOverloadMethods {
    void process(int x) { }
    void helper() { }
    void process(String s) { }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.5.0-SNAPSHOT-all.jar -c config.xml OrdinaryOverloadMethods.java

Starting audit...
[ERROR] OrdinaryOverloadMethods.java:4:5: All overloaded methods should be placed next to each other. Previous overloaded method located at line '2'. [OverloadMethodsDeclarationOrder]
Audit done.
Checkstyle ends with 1 errors.
```

---

**Describe what you expect in detail.**

OverloadMethodsDeclarationOrderCheck must flag overloaded top-level methods that are not grouped together in a JEP 512 compact source file. The check's `getRequiredTokens` returns only `OBJBLOCK`, and it iterates that node's children looking for `METHOD_DEF` nodes with the same name. A compact source file's implicit class is not represented by an `OBJBLOCK`, so the check never visits the file's top-level methods and overload grouping is never validated.

Expected on the compact input: 1 violation `All overloaded methods should be placed next to each other. Previous overloaded method located at line '1'.` on the second `process(String s)`, identical to the ordinary class above

**Comment by veddarji** [comment]
I am working on this! I have already implemented the fix by adding COMPACT_COMPILATION_UNIT to getRequiredTokens() and handling it in visitToken(). All tests are passing locally. I will open a Pull Request for this now!

**Comment by romani** [comment]
fix is merged
